csi/volumeutils.py

```diff
@@ -1537,3 +1674,64 @@ def yield_list_of_pvcs(max_entries=0):
             pvcs *= 0
     if idx == -1:
         return "No PVC exist in any storage pool"
+
+
+# noqa # pylint: disable=too-many-arguments
+def execute_gluster_quota_command(privkey, user, host, gvolname, path, size):
+    """
+    Function to execute the GlusterFS's quota command on external cluster
+    """
+    # 'size' can always be parsed as integer with no errors
+    size = int(size) * 0.95
+
+    host = reachable_host(host)
+    if host is None:
+        errmsg = "All hosts are not reachable"
+        logging.error(logf(errmsg))
+        return errmsg
+
+    quota_cmd = [
+        "ssh",
+        "-oStrictHostKeyChecking=no",
+        "-i",
+        privkey,
+        f"{user}@{host}",
+        "sudo",
+        "gluster",
+        "volume",
+        "quota",
+        gvolname,
+        "limit-usage",
+        f"/{path}",
+        f"{size}",
+    ]
+    try:
+        execute(*quota_cmd)
+    except CommandException as err:
+        errmsg = "Unable to set Gluster Quota via ssh"
+        logging.error(logf(errmsg, error=err))
+        return errmsg
+
+    return None
+
+
+def upgraded_volume_context(ctx):
+    """
+    Upgrade the Volume Context with propper field names
+    """
+    # Already using the upgraded version
+    if ctx.get("version", "1") >= LATEST_VOLUME_CONTEXT_VERSION:
```

Comment:
File "/kadalu/volumeutils.py", line 1723, in upgraded_volume_context
    if ctx.get("version", "1") >= LATEST_VOLUME_CONTEXT_VERSION:
TypeError: '>=' not supported between instances of 'str' and 'int'